### PR TITLE
ramips: Add suport for COMFAST CF-WR617AC

### PR DESCRIPTION
**Pull request by tsipa**
COMFAST CF-WR617AC is a dual-radio wireless router with a 4 fast ethernet port.
It has 4*5dBi antennas

Specifications:
Chipset:MT7628DA+MT7612E
Antenna : 2.4Ghz:2*5dbi Antenna + 5.8Ghz:2*5dbi Antenna 
Wireless Rate:2.4Ghz 300Mbps , 5.8Ghz 867Mbps 
Output Power :100mW(20dbm)
Physical port:1*10/100Mbps RJ45 WAN Port , 3*10/100Mbps RJ45 LAN Port 

Signed-off-by: Sergei Iudin <tsipa740@gmail.com>

### CODE REVIEW

**Comment by tsipa** [comment]
Not really sure if i can make it to get PR merged. Few months ago i have managed to solve my problem with PR of some random guy(whose PR didn't get merged, lol), so would like to put some minimal effort return favor to the project.

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  model = "Comfast CF-WR617AC";
+
+  chosen {
+    bootargs = "console=ttyS0,115200";
+  };
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x4000000>;
+	};
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:16]: Memory node can be dropped.

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+#include "mt7628an.dtsi"
+
+/ {
+  compatible = "comfast,cf-wr617ac", "mediatek,mt7628an-soc";
+  model = "Comfast CF-WR617AC";
+
+  chosen {
+    bootargs = "console=ttyS0,115200";
+  };
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:11]: Please take take of the indent, DTS should be tabs-only.

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+/dts-v1/;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:1]: Add an SPDX license identifier.

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x4000000>;
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:22]: flash@0

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:25]: Consider using a higher frequency.

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			ubootenv: partition@30000 {
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:38]: This DT label (ubootenv:) doesn't seem to be needed?

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0x760000>;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:53]: So, what about 0x7a0000 till the end of the flash?

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+			};
+		};
+	};
+};
+
+
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:59]: No double empty lines.

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    compatible = "mediatek,mt76";
+    reg = <0x0000 0 0 0 0>;
+    mediatek,mtd-eeprom = <&factory 0x8000>;
+    ieee80211-freq-limit = <5000000 6000000>;
+  };
+};
+
+&ethernet {
+  mtd-mac-address = <&factory 0x4>;
+  mtd-mac-address-increment = <(-1)>;
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:75]: Please check whether there are addresses in 0x28, 0x2e, 0xe000 and 0xe006.

FYI: https://openwrt.org/docs/guide-developer/mac.address#find_out_about_flash_locations

File: target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+&esw {
+  mediatek,portmap = <0x2f>;
+  mediatek,portdisable = <0x2a>;
+};
+
+&wmac {
+  status = "okay";
+};
+
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7628an_comfast_cf-wr617ac.dts:86]: No empty line at EOF.

Does this really have no LEDs or buttons?

File: target/linux/ramips/image/mt76x8.mk [hunk 11]
```diff
@@ -485,3 +485,14 @@ define Device/zyxel_keenetic-extra-ii
 	check-size $$$$(IMAGE_SIZE) | zyimage -d 6162 -v "ZyXEL Keenetic Extra II"
 endef
 TARGET_DEVICES += zyxel_keenetic-extra-ii
+
+define Device/comfast_cf-wr617ac
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:489]: Please move up to maintain alphabetic sorting.

File: target/linux/ramips/image/mt76x8.mk [hunk 12]
```diff
@@ -485,3 +485,14 @@ define Device/zyxel_keenetic-extra-ii
...
+  IMAGE_SIZE := $(ralink_default_fw_size_32M)
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:490]: That's 1. not used anymore and 2. wrong.

File: target/linux/ramips/image/mt76x8.mk [hunk 13]
```diff
@@ -485,3 +485,14 @@ define Device/zyxel_keenetic-extra-ii
...
+  DTS := CF-WR617AC
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:491]: This line can be dropped if the DTS with long name is used.

File: target/linux/ramips/image/mt76x8.mk [hunk 14]
```diff
@@ -485,3 +485,14 @@ define Device/zyxel_keenetic-extra-ii
...
+  DEVICE_VENDOR := Comfast
+  DEVICE_MODEL := CF-WR617AC
+  DEVICE_TITLE := Comfast CF-WR617AC
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:494]: This line can be dropped.

File: target/linux/ramips/image/mt76x8.mk [hunk 15]
```diff
@@ -485,3 +485,14 @@ define Device/zyxel_keenetic-extra-ii
...
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-rt2800-pci
+  SUPPORTED_DEVICES += cf-wr617ac
```
Comment (adrianschmutzler) [inline target/linux/ramips/image/mt76x8.mk:496]: This line can be dropped as well.

File: target/linux/ramips/base-files/etc/board.d/02_network [hunk 16]
```diff
@@ -471,6 +471,10 @@ ramips_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"4:lan:1" "2:lan:2" "0:wan" "6@eth0"
 		;;
+  comfast,cf-wr617ac)
+    ucidef_add_switch "switch0" \
+      "1:lan:3" "2:lan:2" "3:lan:1" "4:wan" "6@eth0"
+    ;;
```
Comment (adrianschmutzler) [inline target/linux/ramips/base-files/etc/board.d/02_network:477]: Indent and sorting!

**Comment by adrianschmutzler** [comment]
You have the DTS file twice, you will only need one. Despite, some formal issues for the commit itself:
- lower-case "add" in title
- suport -> support
- Use real name for Signed-off-by
- Add commit message with specifications and flashing instructions

Further reading:
https://openwrt.org/submitting-patches

**Comment by adrianschmutzler** [comment]
Just found that this is for openwrt-19.07. We only accept patches for master.

Please implement the feedback and resubmit based on master.

**Pull request closed without merging**